arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -176,11 +176,11 @@ int HttpCommTask<T>::on_header_complete(llhttp_t* p) {
     char const* response = "HTTP/1.1 100 Continue\r\n\r\n";
     auto buff = asio_ns::buffer(response, strlen(response));
     asio_ns::async_write(self->_protocol->socket, buff,
-                         [self](asio_ns::error_code const& ec, std::size_t transferred) {
-                           llhttp_resume(&self->_parser);
-                           self->asyncReadSome();
+                         [self = self->shared_from_this()](asio_ns::error_code const& ec, std::size_t transferred) {
+                           HttpCommTask<T>* task = static_cast<HttpCommTask<T>*>(self.get());
+                           task->asyncReadSome();
```

Comment:
this seems wrong

Reply:
why do you think it is wrong? please clarify

Reply:
calling `asyncReadSome()` should be redundant if you return HPE_OK